core/controllers/base.py

```diff
@@ -799,7 +799,10 @@ def _create_token(cls, user_id, issued_on):
         # Round time to seconds.
         issued_on = str(int(issued_on))
 
-        digester = hmac.new(CSRF_SECRET.value.encode('utf-8'))
+        digester = hmac.new(
+            key=CSRF_SECRET.value.encode('utf-8'),
+            digestmod='md5'
+        )
```

Comment:
Why did you need to change this?

Reply:
The digestmod parameter previously had an implicit / default value of 'md5' as the default digest constructor. But after i tried to migrate to new python version, it generated error for `missing argument value for digestmod`. Since we were using the implicit type 'md5' before, I just passed the md5 constructor value to it in string form. We can use string value or something like hashlib.md5(). String one worked for me here.